File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/tbdex/pfi/PfiOnboardingTest.kt
package website.tbd.developer.site.docs.tbdex.pfi

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions.*

// :snippet-start: pfiOnboardingImportsKt
import web5.sdk.dids.methods.dht.CreateDidDhtOptions
import web5.sdk.dids.methods.dht.DidDht
import web5.sdk.dids.didcore.Service
// :snippet-end:
import web5.sdk.crypto.InMemoryKeyManager

/**
 * Tests for PFI Onboarding guide
 */
class PfiOnboardingTest {

  @Test
  fun `create PFI DID`() {
    val keyManager = InMemoryKeyManager()
    // :snippet-start: pfiOnboardingCreateDidKt
    val serviceToAdd = Service.Builder()
        .id("pfi")
        .type("PFI")
        .serviceEndpoint(listOf("https://example.com/"))
        .build()

    val options = CreateDidDhtOptions(
        publish = true,
        services = listOf(serviceToAdd),
    )

    val pfiDid = DidDht.create(keyManager, options)
    // :snippet-end:

    assertEquals("PFI", pfiDid.document.service?.get(0)?.type, "DID should start with 'did:dht'")
  }
}

File: site/testsuites/testsuite-javascript/__tests__/tbdex/pfi/pfiOnboarding.test.js
// :snippet-start: pfiOnboardingImportsJs
import { DidDht } from '@web5/dids';
// :snippet-end:
import { LocalKeyManager } from '@web5/crypto';
import { test, expect, describe } from 'vitest';

describe('PFI: Onboarding', () => {
    test('Create DID for a PFI', async () => {
        const keyManager = new LocalKeyManager();
        // :snippet-start: pfiOnboardingCreateDidJs
        const pfiDid = await DidDht.create({
            options:{
                publish: true,
                services: [{
                    id: 'pfi',
                    type: 'PFI',
                    serviceEndpoint: 'https://example.com/'
                }]
            },
            keyManager
        })

        // :snippet-end:
        expect(pfiDid.document).toHaveProperty('service');
    });
  });